crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs

```diff
@@ -650,12 +651,30 @@ impl Fold for AssignFolder {
                                 elems.len()
                             }),
                         );
-
+                        let r = match *right {
+                            Expr::Ident(Ident {
+                                sym: js_word!("arguments"),
+                                ..
+                            }) => {
+                                if self.c.loose {
+                                    *right
+                                } else {
+                                    Expr::Call(CallExpr {
```

Comment:
This seems inefficient.
Instead of creating a new box, I think creating `Box` here will be better.